src/main/java/org/candlepin/subscriptions/marketplace/MarketplacePayloadMapper.java

```diff
@@ -128,18 +128,26 @@ protected boolean isSnapshotPAYGEligible(TallySnapshot snapshot) {
         var eligibleSnapshots = tallySummary.getTallySnapshots().stream()
             .filter(this::isSnapshotPAYGEligible).collect(Collectors.toList());
 
-        List<UsageEvent> events = new ArrayList<>();
+        // Abandon ship if the usageKey isn't for one that's relevant to us, otherwise we get a whole
+        // bunch of missing subscription spam
+        List<TallySnapshot> premiumProductionSnapshots = eligibleSnapshots.stream().filter(x -> {
+            ServiceLevel sla = ServiceLevel.fromString(x.getSla().toString());
+            Usage usage = Usage.fromString(x.getUsage().toString());
+            return Objects.equals(ServiceLevel.PREMIUM, sla) && Objects.equals(Usage.PRODUCTION, usage);
+        }).collect(Collectors.toList());
```

Comment:
Let's not do this. When all the other bits are sorted out (mainly the billing_type bits) we shouldn't be receiving data for non-Premium or non-Production usage, and if something is wrong and we do receive that data, it'd be best to be alerted about it. Additionally, future requirements may introduce non-Premium or non-Production PAYG.